crates/cargo-util/src/paths.rs

```diff
@@ -193,9 +211,7 @@ pub fn write_if_changed<P: AsRef<Path>, C: AsRef<[u8]>>(path: P, contents: C) ->
         let mut orig = Vec::new();
         f.read_to_end(&mut orig)?;
         if orig != contents {
-            f.set_len(0)?;
-            f.seek(io::SeekFrom::Start(0))?;
-            f.write_all(contents)?;
```

Comment:
Why is this changed?

Reply:
Poor self-control :) Will back out.

Reply:
I mean it might be reasonable (in its own commit); I'd just want to see some investigation as to why its written the way it is.  People don't write code like that out of nowhere.